src/dune_rules/super_context.ml

```diff
@@ -2,13 +2,17 @@ open! Dune_engine
 open! Stdune
 open Import
 
-let default_context_flags (ctx : Context.t) =
-  (* TODO DUNE3 To ensure full backward compatibility, ocaml_cflags are still
-     present in the :standard set of flags. However these should not as they are
-     already prepended when calling the compiler, causing flag duplication. *)
-  let c = Ocaml_config.ocamlc_cflags ctx.ocaml_config in
+let default_context_flags (ctx : Context.t) ~project =
+  let cflags = Ocaml_config.ocamlc_cflags ctx.ocaml_config in
   let cxx =
-    List.filter c ~f:(fun s -> not (String.is_prefix s ~prefix:"-std="))
+    List.filter cflags ~f:(fun s -> not (String.is_prefix s ~prefix:"-std="))
+  in
+  let c =
+    if not (Dune_project.always_add_cflags project) then
+      (* TODO DUNE3 make this the default behavior *)
```

Comment:
Is this TODO necessary? If this default behavior, this would be reflected by `always_add_cflags` already.